processing/src/main/java/io/druid/segment/filter/AndFilter.java

```diff
@@ -42,10 +43,9 @@
 
   private final List<Filter> filters;
 
-  public AndFilter(
-      List<Filter> filters
-  )
+  public AndFilter(List<Filter> filters)
   {
+    Preconditions.checkArgument(filters.size() > 0, "Can't construct empty AndFilter");
```

Comment:
This change was suggested by @jihoonson here: https://github.com/druid-io/druid/pull/4079#discussion_r109273193 and I agree with it, because `AndFilter` is either created from `AndDimFilter`, which prohibits empty list too, or as a result of internal transformations of existing AndFilters and OrFilters. If an empty list passed, it is a bug in Druid's code and it's better to throw exception than silently consume the bug.